packages/DataGrid/src/DataGrid.js

```diff
@@ -0,0 +1,628 @@
+        return ColumnDefinition.props.isSticky ? index : null;
+      }).filter(chunk => chunk !== null),
+    [ColumnDefinitions]
+  );
+
+  const stickyGridWidth = React.useMemo(() => {
+    let width = 0;
+    if (stickyColumnsIndexes.length) {
+      stickyColumnsIndexes.forEach(index => {
+        width += ColumnDefinitions[index].props.width;
+      });
+    }
+    return width;
+  }, [ColumnDefinitions, stickyColumnsIndexes]);
+
+  const highlightRow = React.useCallback(({ rowIndex = null }) => {
+    if (rowIndex !== null) {
+      if (refPrevActiveRow.current !== null) {
+        const rowIndex = refPrevActiveRow.current;
+        const key = `0${rowIndex}`;
+        if (refsCell.current.keys[key]) refsCell.current.keys[key].deemphasizeOnrow(rowIndex);
+      }
+
+      refActiveRow.current = rowIndex;
+      refPrevActiveRow.current = rowIndex;
+
+      // first column nth row only
+      const key = `0${rowIndex}`;
+      if (refsCell.current.keys[key]) refsCell.current.keys[key].highlightOnRow(rowIndex);
+    }
+  }, []);
+
+  const deemphasizeRow = React.useCallback(() => {
+    if (refPrevActiveRow.current) {
+      const rowIndex = refPrevActiveRow.current;
+
+      // first column nth row only
+      const key = `0${rowIndex}`;
+      if (refsCell.current.keys[key]) refsCell.current.keys[key].deemphasizeOnrow(rowIndex);
+    }
+
+    refActiveRow.current = null;
+    refPrevActiveRow.current = null;
+  }, []);
+
+  const onChangeActiveCell = React.useCallback(({ columnIndex, rowIndex }) => {
+    const key = `${columnIndex}${rowIndex}`;
+
+    if (refPrevActiveCell.current && refPrevActiveCell.current in refsCell.current.keys) {
+      const prevCell = refsCell.current.keys[refPrevActiveCell.current];
+      if (prevCell) {
+        refsCell.current.keys[refPrevActiveCell.current].isActiveCell(false);
+      }
+    }
+
+    refPrevActiveCell.current = key;
+    refsCell.current.keys[key].isActiveCell(true);
+  }, []);
+
+  const { handleKeyDown, handleKeyUp, handleClick, gridId, restoreHighlightFocus } = useGridEventHandler({
+    columnCount,
+    highlightRow,
+    onChangeActiveCell,
+    onClick,
+    onEnter,
+    onKeyDown,
+    onRowChecked,
+    onShiftSpaceBar,
+    onSpaceBar,
+    refContainer,
+    refGrid,
+    rowCount,
+    rowHeight,
+    scrollBarWidth,
+    stickyColumnsIndexes,
+  });
+
+  const a11yTextMessage = React.useCallback((value, column, rowIndex) => {
+    return `${value}. You are on row ${rowIndex}, column ${column}. Disregard the following information:`;
+  }, []);
+
+  const handleScroll = React.useCallback(parameters => {
+    const { scrollLeft, scrollTop /* scrollUpdateWasRequested */ } = parameters;
+
+    if (refScrollHeader.current) {
+      refScrollHeader.current.scrollTo({ left: scrollLeft, top: 0 });
+    }
+
+    // prevent rescrolling when this scrollbar gets sync with the one in the sticky column
+    if (refScrollHappenedBy.current === null) {
```

Comment:
If it were me, i'd reverse the logic of this `if` block. Same with the one on line 261.

Reply:
won't do.